src/Gizmos/axisDragGizmo.ts

```diff
@@ -74,41 +72,44 @@ export class AxisDragGizmo extends Gizmo {
      * @param dragAxis The axis which the gizmo will be able to drag on
      * @param color The color of the gizmo
      */
-    constructor(dragAxis: Vector3, color: Color3 = Color3.Gray(), gizmoLayer: UtilityLayerRenderer = UtilityLayerRenderer.DefaultUtilityLayer, parent: Nullable<PositionGizmo> = null) {
-        super(gizmoLayer);
-        this._parent = parent;
-        // Create Material
-        this._coloredMaterial = new StandardMaterial("", gizmoLayer.utilityLayerScene);
-        this._coloredMaterial.diffuseColor = color;
-        this._coloredMaterial.specularColor = color.subtract(new Color3(0.1, 0.1, 0.1));
-
-        this._hoverMaterial = new StandardMaterial("", gizmoLayer.utilityLayerScene);
-        this._hoverMaterial.diffuseColor = color.add(new Color3(0.3, 0.3, 0.3));
+    constructor(
+        dragAxis: Vector3,
+        color: Color3 = Color3.Gray(),
+        gizmoLayer: UtilityLayerRenderer = UtilityLayerRenderer.DefaultUtilityLayer,
+        parent: Nullable<PositionGizmo> = null
+    ) {
+        super({ dragAxis }, color, gizmoLayer, parent);
 
         // Build mesh on root node
-        this._arrow = AxisDragGizmo._CreateArrow(gizmoLayer.utilityLayerScene, this._coloredMaterial);
+        this._arrow = AxisDragGizmo._CreateArrow(gizmoLayer.utilityLayerScene);
 
         this._arrow.lookAt(this._rootMesh.position.add(dragAxis));
         this._arrow.scaling.scaleInPlace(1 / 3);
         this._arrow.parent = this._rootMesh;
 
+        this._materialSwitcher.registerMeshes(
+            this._rootMesh.getChildMeshes(false)
+        );
+
         var currentSnapDragDistance = 0;
         var tmpVector = new Vector3();
         var tmpSnapEvent = { snapDistance: 0 };
-        // Add drag behavior to handle events when the gizmo is dragged
-        this.dragBehavior = new PointerDragBehavior({ dragAxis: dragAxis });
-        this.dragBehavior.moveAttached = false;
-        this._rootMesh.addBehavior(this.dragBehavior);
 
         var localDelta = new Vector3();
         var tmpMatrix = new Matrix();
         this.dragBehavior.onDragObservable.add((event) => {
             if (this.attachedMesh) {
                 // Convert delta to local translation if it has a parent
                 if (this.attachedMesh.parent) {
-                    this.attachedMesh.parent.computeWorldMatrix().invertToRef(tmpMatrix);
+                    this.attachedMesh.parent

```

Comment:
Keep current code formatting

Reply:
What formatting setup are you using not to defer to doing that manually?
The changes were introduced by VS Code auto-formatting fit Prettier with `"prettier.tabWidth": 4`.
If I disable Prettier, I am not getting much help from formatter.

Reply:
You can do a lint pass
```
gulp tsLint
```

Reply:
That does not revert back to the previous formatting so I'll do that manually and avoid using prettier while coding.

Reply:
Done